plugins/woocommerce-blocks/assets/js/atomic/utils/register-block-single-product-template.ts

```diff
@@ -102,17 +105,14 @@ export const registerBlockSingleProductTemplate = ( {
 	}, 'core/edit-site' );
 
 	subscribe( () => {
-		const isBlockRegistered = Boolean( variationName )
+		const isRegistered = Boolean( variationName )
```

Comment:
I'm thinking that maybe we should skip the logic of `blocksRegistered` Set completely for blocks and just keep if only for variations.

```suggestion
	subscribe( () => {
		const isRegistered = Boolean( variationName )
			? blocksRegistered.has( variationName )
			: isBlockRegistered( blockName );
		// This subscribe callback could be invoked with the core/blocks store
		// which would cause infinite registration loops because of the `registerBlockType` call.
		// This local cache helps prevent that.
		if ( ! isRegistered && isAvailableOnPostEditor && ! editSiteStore ) {
			if ( isVariationBlock ) {
				blocksRegistered.add( variationName ); // keep it for variation
				registerBlockVariation(
					blockName,
					blockSettings as BlockVariation< BlockAttributes >
				);
			} else {
			    // remove the locksRegistered.add( blockName ) for blocks
				// @ts-expect-error: `registerBlockType` is typed in WordPress core
				registerBlockType( blockMetadata, blockSettings );
			}
		}
	}, 'core/edit-post' );
```

GitHub doesn't allow me to put suggestion to the whole block, hence it's not the best diff, sorry about that.

Reply:
I thought the same thing, but I wasn't sure that it made sense! Thanks for your feedback! I implemented it with [34120ec](https://github.com/woocommerce/woocommerce/pull/53217/commits/34120ecfc48efded6f94d54c19f0dd6d66961ae9). Can you take another look?